Sanity workflow collection installation fails on Ansible 2.9

We are seeing a new failure that seems to be something that needs be fixed by the Ansible support group.

### Workflow:

The failure occurs when executing the following workflow:

```
jobs:
  sanity:
    uses: ansible-network/github\_actions/.github/workflows/sanity.yml@main
```

### Failure: (for Ansible core version 2.9)

Run ansible-galaxy collection install ./dellemc-enterprise_sonic-2.2.0.tar.gz --pre -p /home/runner/collections
ansible-galaxy collection install ./dellemc-enterprise_sonic-2.2.0.tar.gz --pre -p
*
*
*
ansible-galaxy: error: unrecognized arguments: --pre

### Details:
Run ansible-galaxy collection install ./dellemc-enterprise_sonic-2.2.0.tar.gz --pre -p /home/runner/collections
ansible-galaxy collection install ./dellemc-enterprise_sonic-2.2.0.tar.gz --pre -p /home/runner/collections
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
PY_COLORS: 1
source_directory: ./source
pythonLocation: /opt/hostedtoolcache/Python/3.7.17/x64
PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.7.17/x64/lib/pkgconfig
Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.7.17/x64
Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.7.17/x64
Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.7.17/x64
LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.7.17/x64/lib:/opt/hostedtoolcache/Python/3.9.17/x64/lib
usage: ansible-galaxy [-h] [--version] [-v] TYPE ...
ansible-galaxy: error: unrecognized arguments: --pre
Error: Process completed with exit code 2.

Sanity for Ansible 2.9 test matrix entries fails as shown above.

Will someone be deleting Ansible 2.9 soon from the test matrix in "https://github.com/ansible-network/github_actions/blob/main/.github/workflows/sanity.yml" since it has reached EOL?

--- a/.github/actions/build_install_collection/action.yml
+++ b/.github/actions/build_install_collection/action.yml
@@ -18,7 +18,11 @@ inputs:
       The collection tarball when built
       If not set, this will be determined by the action
     default: ""
-
+  ansible_version:
+    description: |
+      Ansible Core version from the workflow.
+    required: false
+    default: ""
 outputs:
   collection_path:
     description: The final collection path
@@ -62,6 +66,7 @@ runs:
 
     - name: Install collection python requirements
       if: ${{ inputs.install_python_dependencies == 'true' }}
+
       run: python3 -m pip install -r requirements.txt -r test-requirements.txt
       shell: bash
       working-directory: ${{ inputs.source_path }}
@@ -78,10 +83,17 @@ runs:
       shell: bash
       working-directory: ${{ inputs.source_path }}
 
-    - name: Install collection and dependencies
+    - name: Install collection and dependencies (with --pre flag)
       run: ansible-galaxy collection install ./${{ steps.identify.outputs.tar_file || inputs.tar_file }} --pre -p /home/runner/collections
       shell: bash
       working-directory: ${{ inputs.source_path }}
+      if: ${{ inputs.ansible_version != "stable-2.9" }}
+
+    - name: Install collection and dependencies (without --pre flag)
+      run: ansible-galaxy collection install ./${{ steps.identify.outputs.tar_file || inputs.tar_file }} -p /home/runner/collections
+      shell: bash
+      working-directory: ${{ inputs.source_path }}
+      if: ${{ inputs.ansible_version == "stable-2.9" }}
 
     - name: Copy the galaxy.yml from source to destination, needed for pytest-ansible-units
       run: cp galaxy.yml ${{ steps.identify.outputs.collection_path || inputs.collection_path }}/galaxy.yml

--- a/.github/actions/build_install_collection/action.yml
+++ b/.github/actions/build_install_collection/action.yml
@@ -87,13 +87,13 @@ runs:
       run: ansible-galaxy collection install ./${{ steps.identify.outputs.tar_file || inputs.tar_file }} --pre -p /home/runner/collections
       shell: bash
       working-directory: ${{ inputs.source_path }}
-      if: ${{ inputs.ansible_version != "stable-2.9" }}
+      if: ${{ inputs.ansible_version != 'stable-2.9' }}
 
     - name: Install collection and dependencies (without --pre flag)
       run: ansible-galaxy collection install ./${{ steps.identify.outputs.tar_file || inputs.tar_file }} -p /home/runner/collections
       shell: bash
       working-directory: ${{ inputs.source_path }}
-      if: ${{ inputs.ansible_version == "stable-2.9" }}
+      if: ${{ inputs.ansible_version == 'stable-2.9' }}
 
     - name: Copy the galaxy.yml from source to destination, needed for pytest-ansible-units
       run: cp galaxy.yml ${{ steps.identify.outputs.collection_path || inputs.collection_path }}/galaxy.yml

--- a/.github/workflows/integration.yml
+++ b/.github/workflows/integration.yml
@@ -117,6 +117,7 @@ jobs:
           source_path: ${{ env.source_directory }}
           collection_path: ${{ steps.identify.outputs.collection_path }}
           tar_file: ${{ steps.identify.outputs.tar_file }}
+          ansible_version: ${{ matrix.ansible-version }}
 
       - name: Print the ansible version
         run: ansible --version

--- a/.github/workflows/integration_simple.yml
+++ b/.github/workflows/integration_simple.yml
@@ -123,6 +123,7 @@ jobs:
           source_path: ${{ env.source_directory }}
           collection_path: ${{ steps.identify.outputs.collection_path }}
           tar_file: ${{ steps.identify.outputs.tar_file }}
+          ansible_version: ${{ matrix.ansible-version }}
 
       - name: Print the ansible version
         run: ansible --version

--- a/.github/workflows/sanity.yml
+++ b/.github/workflows/sanity.yml
@@ -150,7 +150,7 @@ jobs:
 
       - name: Pre install collections dependencies first so the collection install does not
         run: ansible-galaxy collection install --pre ${{ inputs.collection_pre_install }} -p /home/runner/collections/
-        if: inputs.collection_pre_install != ''
+        if: ${{ inputs.collection_pre_install != '' && matrix.ansible-version != 'stable-2.9' }}
 
       - name: Read collection metadata from galaxy.yml
         id: identify
@@ -165,6 +165,7 @@ jobs:
           source_path: ${{ env.source_directory }}
           collection_path: ${{ steps.identify.outputs.collection_path }}
           tar_file: ${{ steps.identify.outputs.tar_file }}
+          ansible_version: ${{ matrix.ansible-version }}
 
       - name: Print the ansible version
         run: ansible --version

--- a/.github/workflows/unit_galaxy.yml
+++ b/.github/workflows/unit_galaxy.yml
@@ -143,7 +143,7 @@ jobs:
 
       - name: Pre install collections dependencies first so the collection install does not
         run: ansible-galaxy collection install --pre ${{ inputs.collection_pre_install }} -p /home/runner/collections
-        if: inputs.collection_pre_install != ''
+        if: ${{ inputs.collection_pre_install != '' && matrix.ansible-version != 'stable-2.9' }}
 
       - name: Read collection metadata from galaxy.yml
         id: identify
@@ -158,6 +158,7 @@ jobs:
           source_path: ${{ env.source_directory }}
           collection_path: ${{ steps.identify.outputs.collection_path }}
           tar_file: ${{ steps.identify.outputs.tar_file }}
+          ansible_version: ${{ matrix.ansible-version }}
 
       - name: Print the ansible version
         run: ansible --version

--- a/.github/workflows/unit_source.yml
+++ b/.github/workflows/unit_source.yml
@@ -104,6 +104,7 @@ jobs:
           source_path: ${{ env.source_directory }}
           collection_path: ${{ steps.identify.outputs.collection_path }}
           tar_file: ${{ steps.identify.outputs.tar_file }}
+          ansible_version: ${{ matrix.ansible-version }}
 
       - name: Print the ansible version
         run: ansible --version

